src/devices/acurite.c

```diff
@@ -468,17 +470,23 @@ static int acurite_txr_callback(bitbuffer_t *bitbuf) {
 	    sensor_status = bb[2]; // @todo, uses parity? & 0x07f
 	    humidity = acurite_getHumidity(bb[3]);
 	    tempc = acurite_txr_getTemp(bb[4], bb[5]);
-	    tempf = celsius2fahrenheit(tempc);
-
-	    printf("%s Acurite tower sensor 0x%04X Ch %c: %3.1F C %3.1F F %d %% RH\n",
-		   time_str, sensor_id, channel, tempc, tempf, humidity);
-
-	    // currently 0x44 seens to be a normal status and/or type
-	    // for tower sensors.  Battery OK/Normal == 0x40
-	    if (sensor_status != 0x44)
-		printf("%s Acurite tower sensor 0x%04X Ch %c, Status %02X\n",
-		       time_str, sensor_id, channel, sensor_status);
+            sprintf(channel_str, "%c", channel);
+            battery_low = sensor_status >>7;
+		
+        data = data_make(
+                "time",			"",		DATA_STRING,	time_str,
+                "model",		"",		DATA_STRING,	"Acurite tower sensor",
+                "id",			"",		DATA_INT,	sensor_id,
+                "channel",  		"",     	DATA_STRING, 	&channel_str,
+                "temperature_C", 	"Temperature",	DATA_FORMAT,	"%.1f C", DATA_DOUBLE, tempc,
+                "humidity",		"Humidity",	DATA_INT,	humidity,
+                "battery",        	"Battery",    	DATA_STRING, 	battery_low ? "LOW" : "OK",
+
+                NULL);
+
+        data_acquired_handler(data);
 
+        return 1;
```

Comment:
No, this is wrong.   It shouldn't just return here.   

1. There are possibly still unprocessed rows in the the bitbuffer.
2. IIRC the return value is supposed to be the number of events processed, though I don't remember if anything is doing anything with this. 

Also, the indentation is now wrong.